### DISCUSSION TITLE
Error installing plugin file

### DISCUSSION THREAD

**Discussion by MarieClaess** [neuropoly/axondeepseg#673] [Help needed]
Hi, 

When opening FSLEYES and installing the plugin file, I get following error: **"An error occurred installing the plugin file. Details: No module named 'AxonDeepSeg'."** 

When I checked it, the module is installed though. Any chance you can offer some help?

Many thanks, 
Marie (KU Leuven, Belgium)

**Comment by mathieuboudreau** [comment]
Hi @MarieClaess ! 

It sounds like you might of had FSLeyes installed previously on your computer, prior to your interest in AxonDeepSeg?

The reason why you might be getting this error is because 1) you're not opening the FSLeyes that's installed during the AxonDeepSeg installation procedure and/or 2) you haven't activated the AxonDeepSeg conda virtual environment.

Could you run `which fsleyes` and `conda info --envs`, and share the output for each?

**Comment by MarieClaess** [comment]
Hi @mathieuboudreau, 

I am working on Windows, so I ran "where fsleyes". I get following outputs with these commands: 

(base) C:\Users\u0118748>conda activate ads_venv

(ads_venv) C:\Users\u0118748>where fsleyes
C:\Workdir\Programs\MiniConda\envs\ads_venv\Scripts\fsleyes.exe

(ads_venv) C:\Users\u0118748>conda info --envs
# conda environments:
#
base                     C:\Workdir\Programs\MiniConda
ads_venv              *  C:\Workdir\Programs\MiniConda\envs\ads_venv
rbpmscount         C:\Workdir\Programs\MiniConda\envs\rbpmscount
rgcode                 C:\Workdir\Programs\MiniConda\envs\rgcode

**Reply by mathieuboudreau** [reply]
Thanks @MarieClaess !

As mentionned in our installation documentation (https://axondeepseg.readthedocs.io/en/latest/documentation.html#install-axondeepseg), the external software FSLeyes does not support the Windows operating system:

<img width="749" alt="182644221-043bc5a3-409c-45e4-8913-10dfdc4c748e" src="https://user-images.githubusercontent.com/1421029/183740077-571ac6c0-8fe2-468c-b7f8-37719d43fc92.png">

However, you should still be able to use AxonDeepSeg to segment images and produce morphometrics files using the command line tools:

https://axondeepseg.readthedocs.io/en/latest/documentation.html#segmentation

https://axondeepseg.readthedocs.io/en/latest/documentation.html#morphometrics

If you want to use the plugin we created for graphical user interface that FSLeyes offers on that computer, you'll either need to create a Linux virtual machine that runs in it (you can look up the tool VirtualBox and some pre-made images for it that are available online), or explore the new FSL documentation on running FSLeyes in the Windows Subsystem for Linux (don't think anyone has tried this using our plugin yet though): https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FslInstallation/Windows#Windows_Subsystem_for_Linux.

**Reply by MarieClaess** [reply]
Thanks @mathieuboudreau for the quick reply!
Sorry for my late response and the fact that I did not know that the software could not be used on Windows. I have been trying to run FSLeyes in the Windows Subsystem for Linux, but so far, it was not successful. But I'll get there. Many thanks for your help!

**Comment by mathieuboudreau** [comment]
@MarieClaess Just out of curiosity, would you mind sharing how you heard about AxonDeepSeg, and what you're interested in using it for?

**Reply by MarieClaess** [reply]
I went to the ISER/Brightfocus meeting in Atlanta and there was a poster presentation in which different automated deep learning methods for assessing optic nerve axons were compared. That sparked my interest, so I went looking for all available programs as we are currently (manually) evaluating TEM optic nerve sections of a mouse model.

**Reply by mathieuboudreau** [reply]
Great - thanks for letting us know!

**Reply by mathieuboudreau** [reply]
Also, as I said in the other comment, you don't *need* to use FSLeyes to run the segmentation model, you should be able to run it on Windows using the command line. Here's the documentation for that: , https://axondeepseg.readthedocs.io/en/latest/documentation.html#segmentation
